packages/infra/lib/hl7v2-stack/vpn.ts

```diff
@@ -0,0 +1,69 @@
+import * as cdk from "aws-cdk-lib";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import { Construct } from "constructs";
+import { Hl7v2VpnConfig } from "../../config/hl7v2-config";
+export interface VpnStackProps extends cdk.NestedStackProps {
+  vpc: ec2.IVpc;
+  vpnConfig: Hl7v2VpnConfig;
+}
+
+export class VpnStack extends cdk.NestedStack {
+  constructor(scope: Construct, id: string, props: VpnStackProps) {
+    super(scope, id, props);
+
+    // 1. Create a Virtual Private Gateway (VGW)
+    const vpnGateway = new ec2.CfnVPNGateway(this, "VpnGateway", {
+      type: "ipsec.1",
+      amazonSideAsn: 65000, // or remove to let AWS pick a default
+    });
+
+    // Attach the VGW to your VPC
+    new ec2.CfnVPCGatewayAttachment(this, "VpcVpnAttachment", {
+      vpcId: props.vpc.vpcId,
+      vpnGatewayId: vpnGateway.ref,
+    });
+
+    // 2. Create a Customer Gateway
+    const customerGateway = new ec2.CfnCustomerGateway(this, "CustomerGateway", {
+      bgpAsn: props.vpnConfig.bgpAsn,
+      ipAddress: props.vpnConfig.customerGatewayPublicIp,
+      type: "ipsec.1",
+    });
+
+    // 3. Create the VPN Connection
+    const vpnConnection = new ec2.CfnVPNConnection(this, "VpnConnection", {
+      type: "ipsec.1",
+      vpnGatewayId: vpnGateway.ref,
+      customerGatewayId: customerGateway.ref,
+      staticRoutesOnly: props.vpnConfig.staticRoutesOnly,
+      vpnTunnelOptionsSpecifications: [
+        {
+          preSharedKey: props.vpnConfig.preSharedKeyTunnel1,
+          ikeVersions: [{ value: "ikev2" }],
+          phase1LifetimeSeconds: props.vpnConfig.phase1LifetimeSeconds,
+          phase2LifetimeSeconds: props.vpnConfig.phase2LifetimeSeconds,
+        },
+        {
+          preSharedKey: props.vpnConfig.preSharedKeyTunnel2,
+          ikeVersions: [{ value: "ikev2" }],
+          phase1LifetimeSeconds: props.vpnConfig.phase1LifetimeSeconds,
+          phase2LifetimeSeconds: props.vpnConfig.phase2LifetimeSeconds,
+        },
```

Comment:
Why two tunnels? It'd be good to add a TSDoc to the respective section of configs and/or here.

Reply:
HTX has a failover site to another IP. to be honest its just for document retrieval, but since I believe most if not all of these HIEs will have a failover, I thought we just set up the two tunnels. Is this the right way of doing this?

Reply:
asked for more details on Slack